testing/trino-testing/src/main/java/io/trino/testing/LocalSpoolingManager.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testing;
+
+import io.trino.spi.Plugin;
+import io.trino.spi.protocol.SpooledLocation;
+import io.trino.spi.protocol.SpooledLocation.CoordinatorLocation;
+import io.trino.spi.protocol.SpooledSegmentHandle;
+import io.trino.spi.protocol.SpoolingContext;
+import io.trino.spi.protocol.SpoolingManager;
+import io.trino.spi.protocol.SpoolingManagerContext;
+import io.trino.spi.protocol.SpoolingManagerFactory;
+import jakarta.annotation.PreDestroy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+import static java.util.Objects.requireNonNull;
+
+public class LocalSpoolingManager
```

Comment:
Why do you need this one.
Why not use `FileSystemSpoolingManager` with local filesystem?

Reply:
there is no local filesystem support in the FileSystemSpoolingManager

Reply:
good question, let me see

Reply:
It won't work, adding `trino-spooling-filesystem` to `trino-testing` will introduce circular dependency.

Reply:
<img width="442" alt="Screenshot 2024-09-09 at 12 27 48" src="https://github.com/user-attachments/assets/4a83f1fb-92d6-425a-ba36-da4dc21b220c">

Reply:
Surely managable. Honestly I would prefer this approach, instead of having another implementation of SpoolingManager, as you provide better coverage to classes which are actually used in prod.